### PR TITLE
Update docs analytics and switch to Github Actions

### PR DESCRIPTION
**Pull request by tbekolay**
**Motivation and context:**
We're switching analytics providers for our docs pages. To generate the docs we need to switch to GH Actions, as we have no TravisCI subscription anymore.

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Types of changes:**

- Non-code change (touches things like tests, documentation, build scripts)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [na] I have included a changelog entry.
- [x] I have added tests to cover my changes.
- [x] I have run the test suite locally and all tests passed.

**Still to do:**
- [ ] Update nengo-sphinx-theme to new website styling

### CODE REVIEW

File: nengo/utils/least_squares_solvers.py [hunk 1]
```diff
@@ -258,7 +265,11 @@ def _conjgrad_iters(calcAx, b, x, maxiters=None, rtol=1e-6):
 
 
 class BlockConjgrad(LeastSquaresSolver):
-    """Solve a multiple-RHS least-squares system using block conj. gradient."""
+    """
+    Solve a multiple-RHS least-squares system using block conj.
+
+    gradient.
```
Comment (drasmuss) [inline nengo/utils/least_squares_solvers.py:271]: This got confused by the `conj.`, should all be on one line.

File: nengo/utils/numpy.py [hunk 2]
```diff
@@ -29,12 +26,7 @@ def is_spmatrix(obj):
 
 maxseed = np.iinfo(np.uint32).max
 maxint = np.iinfo(np.int32).max
-
-
-# numpy 1.17 introduced a slowdown to clip, so
-# use nengo.utils.numpy.clip instead of np.clip
-# This has persisted through 1.19 at least
-clip = np.core.umath.clip if LooseVersion(np.__version__) >= "1.17.0" else np.clip
```
Comment (drasmuss) [inline nengo/utils/numpy.py:37]: Might be worth keeping the comment here to explain why we're doing this.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      - v*
+  workflow_dispatch:
+    inputs:
+      debug_enabled:
+        description: Run the build with SSH debugging enabled
+        required: false
+        default: false
+
+env:
+  NUMPY: numpy
```
Comment (drasmuss) [inline .github/workflows/ci.yml:18]: Is this needed?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    steps:
+      - name: Write .pypirc to file
+        run: |
+          echo '${{ secrets.PYPIRC_FILE }}' > ~/.pypirc
+      - uses: actions/checkout@v3
+      - uses: ./actions/setup
+        with:
+          python-version: "3.10"
+      - uses: ./actions/generate-and-check
+      - uses: ./actions/run-script
```
Comment (drasmuss) [inline .github/workflows/ci.yml:140]: These need to be `nengo/nengo-bones/actions/...@main`

File: .gitignore [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
+_build
+bones-scripts/
+build
+dist
 docs/.doctrees
+docs/examples/**/*.py  # For .ipynb examples converted to .py
+htmlcov
+log.txt
+plots/
+wintest.sh
```
Comment (drasmuss) [inline .gitignore:28]: Doesn't really matter, but `wintest.sh` seems like an oddly specific thing to have in our gitignore.

**Review by drasmuss — COMMENTED** [review]
We're still using appveyor here for the windows testing, do we want to switch that to Actions as well?

**Comment by drasmuss** [comment]
We should probably do at least a pretend release-candidate (maybe an actual release?) to check `test-all` and `deploy`.

File: nengo/connection.py [hunk 6]
```diff
@@ -439,6 +441,9 @@ class Connection(NengoObject):
         The slice associated with ``pre`` if it is an ObjView, or None.
     seed : int
         The seed used for random number generation.
+    size_in
+    size_mid
+    size_out
```
[the remark this answers is not in the document]
Comment (tbekolay) [reply nengo/connection.py:446]: Yes, this is apparently a Numpydoc feature for properties and attributes with docstrings: https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring

File: .github/workflows/ci.yml (hunk 3 above)
Comment (tbekolay) [reply .github/workflows/ci.yml:18]: At one point I had to add it to one of the tests to fix an issue, but I might have done that in places where it didn't need it too, I'll check.

File: .gitignore (hunk 5 above)
Comment (tbekolay) [reply .gitignore:28]: I believe that this was for a very old way I tried to do Windows testing using Vagrant. Definitely not relevant anymore, haha

**Comment by drasmuss** [comment]
That was TravisCI that had the security issue where they just publicly exposed your secrets on Windows builds 😬

File: nengo/connection.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (tbekolay) [reply nengo/connection.py:446]: Yeah it renders pretty nicely, links to the property.

**Comment by tbekolay** [comment]
The RC branch is all working after a few small changes, so take a look at the recent fixups @drasmuss and when you do I'll squash down the history and this'll be ready to merge after we update the Sphinx theme.

**Review by drasmuss — APPROVED** [review]

**Pull request merged**